### PR TITLE
adding single file close feature

### PR DESCRIPTION
**Pull request by xalbertoisorna**
Closes: [AP-355](https://xmosjira.atlassian.net/browse/AP-355?atlOrigin=eyJpIjoiZDRiMzQ0YjlkYzFiNDFhNzhjMzJkNGM1ZGQxMTEzZDkiLCJwIjoiaiJ9)

Changes:
- adding xscope_fclose function
- adding tests files using xcommon_cmake
- adding probe: host_close to config.xscope.txt

[AP-355]: https://xmosjira.atlassian.net/browse/AP-355?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Jenkinsfile
Comment (xalbertoisorna) [inline Jenkinsfile:1]: As xcommon_cmake requires the structure xscope_fileio/xscope_fileio we have to include the "dir" command.

File: xscope_fileio/api/xscope_io_device.h [hunk 1]
```diff
@@ -117,6 +117,18 @@ int xscope_ftell(xscope_file_t *xscope_file);
  ******************************************************************************/
 void xscope_close_all_files(void);
 
+
+/******************************************************************************
+ * xscope_fclose
+ *
+ * Closes a single file on the host.
+ * It can be called at any time to close a file.
+ *
```
Comment (ed-xmos) [inline xscope_fileio/api/xscope_io_device.h:126]: It is still necessary to call close_all to terminate the session

File: host/README.rst [hunk 2]
```diff
@@ -0,0 +1,28 @@
+Installing xscope_fileio host tools
```
Comment (ed-xmos) [inline host/README.rst:1]: Nice addition

File: host/README.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    >> Linux and Mac
+    pip install -e ../.
+
+.. tab:: Windows
+
+  .. code-block:: console
+
+    >> Windows
+    pip install -e ../.
+    cmake -G Ninja .
```
Comment (ed-xmos) [inline host/README.rst:25]: I wasn't aware windows needed this

File: tests/close_files/config.xscope [hunk 4]
```diff
@@ -0,0 +1,10 @@
+<xSCOPEconfig ioMode="basic" enabled="true">
```
Comment (ed-xmos) [inline tests/close_files/config.xscope:1]: I sort of think this file should be common. But for another PR..

File: tests/close_files/src/test.c [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+        xscope_file_t fp = xscope_open_file(filename, "wb");
+        data[0] = file_n; // first index of data is the file number
+        xscope_fwrite(&fp, data, BUFF_SIZE);
+        xscope_fclose(&fp);
+    }
+
+}
+
+
+// This test
```
Comment (ed-xmos) [inline tests/close_files/src/test.c:58]: Missing comment?

File: xscope_fileio/src/xscope_io_device.c [hunk 6]
```diff
@@ -195,3 +208,15 @@ void xscope_close_all_files(void){
...
+
+void xscope_fclose(xscope_file_t *xscope_file){
+    xscope_fileio_lock_acquire();
+    const unsigned char idx = xscope_file->index + '0';
+    xscope_bytes(XSCOPE_ID_HOST_CLOSE, 1, &idx);
+    if(VERBOSE) {
+        printf("Sent close file id: %d\n", xscope_file->index);
+    }
+    reset_available_file_idx(xscope_file->index);
+    delay_ticks(10); // sanity time to close file
```
Comment (ed-xmos) [inline xscope_fileio/src/xscope_io_device.c:220]: Be interesting to know where 10 comes from

**Review by ed-xmos — APPROVED** [review]
Looks good.
I note the repo still defaults to main (one for Brennan or Michael) and you need to update https://github.com/xmos/xscope_fileio/blob/develop/README.rst with the new API. In time this should be doxy.
Couple of other small comments (see feedback in github)

File: host/README.rst (hunk 3 above)
Comment (xalbertoisorna) [reply host/README.rst:25]: On wodnwos yes, we could add a build for windows if we assume a compiler on host side in a future version

File: xscope_fileio/src/xscope_io_device.c (hunk 6 above)
Comment (xalbertoisorna) [reply xscope_fileio/src/xscope_io_device.c:220]: in reality is not needed, I was thinking to give a small time to the next file, but in reallity always that th lock is adquired we wont be able to open a file while one is closing on device.

**Pull request merged**